discovery_engine/lib/src/api/events/engine_events.dart

```diff
@@ -185,6 +185,21 @@ class EngineEvent with _$EngineEvent {
   const factory EngineEvent.availableSourcesListRequestFailed() =
       AvailableSourcesListRequestFailed;
 
+  /// Event created as a success response to SetSourcesRequested event.
+  /// Returns both sets of trusted and excluded [Source]s back to the client.
+  @Implements<FeedEngineEvent>()
+  const factory EngineEvent.setSourcesRequestSucceeded({
+    required Set<Source> trustedSources,
+    required Set<Source> excludedSources,
+  }) = SetSourcesRequestSucceeded;
+
+  /// Event created as a success response to SetSourcesRequested event.
+  /// Returns a set of duplicates [Source]s back to the client.
+  @Implements<FeedEngineEvent>()
+  const factory EngineEvent.setSourcesRequestFailed(
```

Comment:
Can this operation then only fail if we have duplicates?
What if it fails for other scenarios?

Reply:
Ah nevermind, then it's an EngineExceptionRaised :)
Saw it below, so ignore this